doc/system-telemetry/reboot-cause.md

```diff
@@ -12,37 +12,43 @@
 ### Enable sonic streaming telemetry agent to send Reboot-cause information
 
 #### Part 1
-During the boot, the process-reboot-cause processes the last reboot-cause based on the hardware reboot-cause
+During the boot, the `determine-reboot-cause`(previously `process-reboot-cause`) determines the last reboot-cause based on the hardware reboot-cause
 and the software reboot-cause information and creates previous-reboot-cause.txt with the information as it does currently.
```

Comment:
Is the previous-reboot-cause.txt file still necessary? It seems redundant now that we will have multiple files in /host/reboot-cause/previous-reboot-cause/. I believe the only thing which reads from this file is the CLI, which will now simply reference the data in State DB.

Reply:
Currently process-reboot-cause is delayed to get the reboot-cause logged into the syslog server so I could see that the returned reboot-cause is wrong until the process-reboot-cause service is running.
The reason to keep and read the file for the last reboot-cause instead of db, the `show reboot-cause` will return the reboot-cause almost immediately after system boots. It's redundant but it can give the consistent output since boot. I was also considering to parse the other saved files to get the last reboot-cause. But I felt also that one additional file to maintain the last reboot-cause won't be that harm.

Reply:
It won't cause harm, but it adds complexity, and potential maintenance. I can understand wanting to make the reboot cause available via the CLI before the DB is populated, but is it necessary? Maybe the `show reboot-cause` command could simply return "Not available yet ... Please try again in a few minutes"? Otherwise, I would prefer the `show reboot-cause` command parses the most recent of the new saved files. This way, there is only one format to deal with.

Also, the `show reboot-cause history` command will also fail until the DB is populated, so it will have the same issue. How do you plan to deal with that?